backend/dataall/cdkproxy/stacks/environment.py

```diff
@@ -378,28 +370,55 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
             parameter_name=f'/dataall/{self._environment.environmentUri}/cfn/custom-resources/lambda/name',
         )
 
-        LakeformationDefaultSettingsProvider = cr.Provider(
-            self,
-            f'{self._environment.resourcePrefix}LakeformationDefaultSettingsProvider',
-            on_event_handler=lf_default_settings_custom_resource,
+        # Data lake location custom resource
+        entry_point = str(
+            pathlib.PosixPath(
+                os.path.dirname(__file__), "../assets/datalakelocationcustomresource"
+            ).resolve()
         )
 
-        default_lf_settings = CustomResource(
+        datalakelocation_cr_dlq = self.set_dlq(
+            f'{self._environment.resourcePrefix}-datalakelocationcr-{self._environment.environmentUri}'
+        )
+        datalake_location_custom_resource = _lambda.Function(
             self,
-            f'{self._environment.resourcePrefix}DefaultLakeFormationSettings',
-            service_token=LakeformationDefaultSettingsProvider.service_token,
-            resource_type='Custom::LakeformationDefaultSettings',
-            properties={
-                'DataLakeAdmins': [
-                    f'arn:aws:iam::{self._environment.AwsAccountId}:role/{self.pivot_role_name}',
-                ]
+            "DatalakeLocationCustomResourceHandler",
+            function_name=f'{self._environment.resourcePrefix}-datalakelocation-handler-{self._environment.environmentUri}',
+            role=self.pivot_role,
+            handler="index.on_event",
+            code=_lambda.Code.from_asset(entry_point),
+            memory_size=1664,
```

Comment:
just a general question, how memory and timeout settings were selected for all this custom resource lambda functions?

Reply:
To my knowledge I think there was no standard way of defining them. Do you have any ideas?

Reply:
I think this code does very simple stuff like calling the API only. So memory and timeout settings seems to be a little bit to high:

There is a way at least I have used to find the best settings: https://github.com/alexcasalboni/aws-lambda-power-tuning

Reply:
https://github.com/awslabs/aws-dataall/issues/382 does it make sense? :)

Reply:
Thank you!!